backend/common/services/cache_service.py

```
@@ -177,6 +177,31 @@ def rpop(self, key: str) -> Optional[Any]:
+        try:
+            result = self.redis.brpop(key, timeout=timeout)
+            if result is None:
+                return None
+            # brpop returns (key, value) tuple
+            _, value = result
+            try:
+                return json.loads(value)
+            except (json.JSONDecodeError, TypeError):
+                return value
+        except RedisError as e:
+            logger.error(f"[REDIS] Error brpop from {key}: {e}")
```

Ruff (0.15.0)

[warning] Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)